### PR TITLE
feat: add os agnostic build system

### PR DESCRIPTION
**Pull request by Cafe137**

### CODE REVIEW

**Comment by vojtechsimetka** [comment]
Can I test this without having windows somehow? Any idea? :)

**Comment by Cafe137** [comment]
> Can I test this without having windows somehow? Any idea? :)

Gave up on it, I built a Windows PC 😄

**Comment by agazso** [comment]
> Can I test this without having windows somehow? Any idea? :)

You may want to try Windows in a VirtualBox, that's what I do.

File: hooks.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    copySync('icon.png', join(options.outputPaths[0], 'icon.png'))
+    copySync('static', join(options.outputPaths[0], 'static'))
+  },
+}
+
+function downloadFile(url, target) {
+  return fetch(url)
+    .then(x => x.arrayBuffer())
+    .then(x => writeFileSync(target, Buffer.from(x)))
+}
```
Comment (vojtechsimetka) [inline hooks.js:42]: I know this is POC but can these be in typescript? Happy to help.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+name: Release
+on:
+  release:
+    types:
+      - created
```
Comment (vojtechsimetka) [inline .github/workflows/release.yml:5]: Should I introduce the release-please action?

Comment (Cafe137) [reply .github/workflows/release.yml:5]: Yes please 🙂 If we can get this workflow in sync with release-please, that would be awesome

File: hooks.js (hunk 1 above)
Comment (Cafe137) [reply hooks.js:42]: Should be fine, it is not very obvious since this script is specified in package.json as a property of forge but I'll check 🙂

Comment (vojtechsimetka) [reply hooks.js:42]: did you check? :)

File: package.json [hunk 3]
```diff
@@ -6,9 +6,9 @@
   "main": "dist/index.js",
   "scripts": {
     "start": "tsc && electron-forge start",
-    "package": "electron-forge package",
-    "make": "electron-forge make",
-    "publish": "electron-forge publish",
+    "package": "tsc && electron-forge package",
+    "make": "tsc && electron-forge make",
+    "publish": "tsc && electron-forge publish",
```
Comment (vojtechsimetka) [inline package.json:11]: I would actually suggest erasing the out folder (I had some issues where after the transpilation with tsc the js files were not updated).

Maybe we can introduces some `clean` script with `rimraf out`

**Review by vojtechsimetka — APPROVED** [review]
LGTM and seems to work on mac

**Pull request merged**